### PR TITLE
Osmnode size reduction

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

We don't want to provision a larger hard drive in AWS due to the increased size of the OSMNode structure that was added for the multi-lingual PR.  During the latest planet data conversion the process failed due to not having enough hard drive space.  This was due to the increased size of the OSMNode structure which in turn increased the OSMWayNode structure.  

Move the linguistic indexes that were added during the multi-lingual PR to a separate structure and save the data to a new sequence file that is not sorted.  Add an index to the OSMNode struct for lookup of the linguistic record.  Refactored graphbuilder and tests as needed.

fixes https://github.com/valhalla/valhalla/issues/4548

Old way_nodes.bin file size: 277248921600
New way_nodes.bin file size: 138012593488

New linguistics_node.bin file size: 4892616

## Tasklist

 - [ ] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Review by nilsnolde — DISMISSED** [review]
thanks a lot for adding some changes so temp files are less in size. I squinted at the code, but have little experience/knowledge on this side of graph building. other than that, looks fine. so again, approving on the premise that the planet conversion was working well and the tests are passing:)

**Comment by gknisely** [comment]
> thanks a lot for adding some changes so temp files are less in size. I squinted at the code, but have little experience/knowledge on this side of graph building. other than that, looks fine. so again, approving on the premise that the planet conversion was working well and the tests are passing:)

Planet looking good.

**Review by dnesbitt61 — APPROVED** [review]

**Review by kevinkreiser — COMMENTED** [review]
i want to suggest a simplification for the new header. give me a minute

File: valhalla/mjolnir/osmnodelinguistic.h [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+  uint32_t ref_pronunciation_nt_sampa_lang_index_;
+  uint32_t ref_pronunciation_katakana_index_;
+  uint32_t ref_pronunciation_katakana_lang_index_;
+  uint32_t ref_pronunciation_jeita_index_;
+  uint32_t ref_pronunciation_jeita_lang_index_;
+};
+} // namespace mjolnir
+} // namespace valhalla
+
+#endif // VALHALLA_MJOLNIR_OSMLINGUISTIC_H
```
Comment (kevinkreiser) [inline valhalla/mjolnir/osmnodelinguistic.h:345]:
```suggestion
#pragma once

#include <cstdint>

namespace valhalla {
namespace mjolnir {

/**
 * OSM node linguistic information.
 */
struct OSMNodeLinguistic {
  static bool is_empty(const OSMNodeLinguistic& maybe_empty) {
    static OSMNodeLinguistic definitely_empty{};
    return maybe_empty == definitely_empty;
  }

  uint32_t name_lang_index_;
  uint32_t name_pronunciation_ipa_index_;
  uint32_t name_pronunciation_ipa_lang_index_;
  uint32_t name_pronunciation_nt_sampa_index_;
  uint32_t name_pronunciation_nt_sampa_lang_index_;
  uint32_t name_pronunciation_katakana_index_;
  uint32_t name_pronunciation_katakana_lang_index_;
  uint32_t name_pronunciation_jeita_index_;
  uint32_t name_pronunciation_jeita_lang_index_;
  uint32_t ref_lang_index_;
  uint32_t ref_pronunciation_ipa_index_;
  uint32_t ref_pronunciation_ipa_lang_index_;
  uint32_t ref_pronunciation_nt_sampa_index_;
  uint32_t ref_pronunciation_nt_sampa_lang_index_;
  uint32_t ref_pronunciation_katakana_index_;
  uint32_t ref_pronunciation_katakana_lang_index_;
  uint32_t ref_pronunciation_jeita_index_;
  uint32_t ref_pronunciation_jeita_lang_index_;
};

} // namespace mjolnir
} // namespace valhalla
```

Comment (kevinkreiser) [reply valhalla/mjolnir/osmnodelinguistic.h:345]: i would humbly ask you to consider a much more succinct implementation for this basic struct to keep it a POD and just assign to and read from the fields directly. in this case zero'ing the struct is a simple as using the brace initializer `{}`. if we upgraded to c++20 we could even have a much more succinct equality check but i guess we'll save that for later

Comment (kevinkreiser) [reply valhalla/mjolnir/osmnodelinguistic.h:345]: also if you'd rather not consider it, no worries

Comment (gknisely) [reply valhalla/mjolnir/osmnodelinguistic.h:345]: Save for later.  ;-)  Unfortunately, I don't have time to retest everything right now.

**Pull request merged**